src/EditorFeatures/Core/ReferenceHighlighting/ReferenceHighlightingViewTaggerProvider.cs

```diff
@@ -69,6 +69,8 @@ protected override ITaggerEventSource CreateEventSource(ITextView textView, ITex
 
         protected override IEnumerable<SnapshotSpan> GetSpansToTag(ITextView textViewOpt, ITextBuffer subjectBuffer)
         {
+            // Note: this may return no snapshot spans.  We have to be resilient to that
+            // when processing the TaggerContext<>.SpansToTag below.
```

Comment:
Or an integration test to cover the actual scenario if we understand it (Dave's repro looks too complicated).